### DISCUSSION TITLE
Second command in chain does not work (syntax/binding)

### DISCUSSION THREAD

**Discussion by tim042849** [i3/i3#4912] [Q&A] [answered]
In my config, I have the following :

`bindsym $wc+u exec palemoon file:///home/tim/Documents/keyboard_shortcuts/html/i3.html; resize shrink width 10 px or 10 ppt`

**Note** `$wc` is a variable that expands to super+control 

If this binding is invoked, only the first command appears to be executed. No error message is noted. 
If the two commands are separated and bound separately, both work.
What is necessary for both commands to run on a single binding?

**Comment by Airblader** [comment, marked as answer]
Are you sure the issue is the second command not being executed at all? A look into the i3 log file would tell you.

The more likely, or at least more severe, issue here is that you are assuming synchronous behavior. i3 will execute both commands in succession, but there is no guarantee whatsoever that by the time the second command is executed, palemoon has opened any window, and thus the second command will most likely end up being executed against the wrong window.

You should be making use of `for_window` to run the second command against palemoon once the window actually exists.

**Reply by tim042849** [reply]
Thank you for the reply AirBladder you have edified me to a degree, but so for I am still not having any luck.
Properties dump for the Pale Moon window:
```
tim@mini:~$ vim /usr/local/bin/i3-get-window-criteria
tim@mini:~$ i3-get-window-criteria
[class="Pale moon" id=54525972 instance="Navigator" title="Pale Moon" window_role="browser"]

Here is one example of what I tried:

`bindsym $wc+u exec palemoon file:///home/tim/Documents/keyboard_shortcuts/html/i3.html; exec for_window class="Pale moon" resize shrink width 10 px or 10 ppt`

The window still does not resize. Using the window title is also unsuccessful

**Edited**: Reversing the order of the commands is also unsuccessful.
Placing the for_window command on a separate line as a stand-alone directive **does** work.


```

**Reply by Airblader** [reply]
`for_window` is not a command, it is a config directive. And to get at least the syntax right please read the user guide.

```
bindsym $wc+u exec palemoon file:///home/tim/Documents/keyboard_shortcuts/html/i3.html
for_window [class="Pale moon"] resize shrink width 10 px or 10 ppt
```

**Reply by tim042849** [reply]
Got it.
